### ISSUE TITLE
ChatGetter overwrites Forward's _client attribute to None

### ISSUE THREAD

**Issue by joell** [LonamiWebs/Telethon#1247]
In `Forward.__init__` the [`self._client` attribute is set to the `client` argument](https://github.com/LonamiWebs/Telethon/blob/13e911957390045a1ad9b0ca7452370b2a7d7f44/telethon/tl/custom/forward.py#L31) *before* `ChatGetter.__init__` is called.

However, `ChatGetter.__init__` sets [`self._client = None`](https://github.com/LonamiWebs/Telethon/blob/13e911957390045a1ad9b0ca7452370b2a7d7f44/telethon/tl/custom/chatgetter.py#L18).

The result of this is that `Forward` objects essentially lose the reference to their creating client after construction.  Attempts to indirectly use the `self._client` attribute later, such as by calling `await my_forward_object.get_chat()` will result in a `NoneType` exception.

I'm not entirely sure why ChatGetter is setting `self._client = None`, but one workaround here might be to set `self._client = client` in `Forward.__init__` *after* the `ChatGetter.__init__` call.

**Comment by Lonami** [comment]
> I'm not entirely sure why ChatGetter is setting `self._client = None`

A instance is often created without having a client yet (it is set later), and creating the attribute later (setting the client without having set it to something on its init) is a worse idea.

**Comment by joell** [comment]
I feel like perhaps I didn't clearly communicate very well at first.  Let me try again.

Please consider the following example:
```python
async def print_forwarder(msg: Message):
    assert msg.forward, "The message must have a forwarder"
    chat = await msg.forward.get_chat()
    print("Message was forwarded from: " + str(chat))
```

If you were to call `print_forwarder` with a `Message` that has a client associated with it, you would expect the `msg.forward.get_chat()` method to work properly and give you information about the associated chat.  However, because `Forward.__init__` calls `ChatGetter.__init__` **after** it sets its `self._client` attribute, that attribute gets replaced by `ChatGetter` with `None`.  So when you call `print_forwarder` in the example above, you will always get an `NoneType` exception thrown from `get_chat()`.  Essentially, `Forward.get_chat()` is broken.

My proposal was to simply move the `self._client` assignment in `Forward.__init__` so that it comes **after**  the `ChatGetter.__init__` call.  Here's a patch illustrating what I'm referring to.

```diff
--- telethon/tl/custom/forward.py.orig	2019-07-30 14:39:02.753015609 -0400
+++ telethon/tl/custom/forward.py	2019-07-30 14:39:26.779682133 -0400
@@ -23,25 +23,26 @@
             Attributes not described here are the same as those available
             in the original :tl:`MessageFwdHeader`.
     """
     def __init__(self, client, original, entities):
         # Copy all the fields, not reference! It would cause memory cycles:
         #   self.original_fwd.original_fwd.original_fwd.original_fwd
         # ...would be valid if we referenced.
         self.__dict__ = dict(original.__dict__)
-        self._client = client
         self.original_fwd = original
 
         sender, input_sender = utils._get_entity_pair(
             original.from_id, entities, client._entity_cache)
 
         if not original.channel_id:
             peer = chat = input_chat = None
         else:
             peer = types.PeerChannel(original.channel_id)
             chat, input_chat = utils._get_entity_pair(
                  utils.get_peer_id(peer), entities, client._entity_cache)
 
         ChatGetter.__init__(self, peer, chat=chat, input_chat=input_chat)
         SenderGetter.__init__(self, original.from_id, sender=sender, input_sender=input_sender)
 
+        self._client = client
+
     # TODO We could reload the message
```

I don't see how this should break anything, and it would fix the `NoneType` exception.  Without some kind of change though, `Forward` objects can never have their `get_chat()` methods called without breaking, which doesn't seem like what was intended.

**Comment by Lonami** [comment]
Thanks for the elaborated report, although this issue was already pretty clear. What I meant by https://github.com/LonamiWebs/Telethon/issues/1247#issuecomment-516538176 is that, when we have a `ChatGetter`, we often don't have a `client` at the moment of creation, it is set later. This is not the case for `Forward`, because when we create it we always have the client.
